prow/crier/reporters/github/reporter.go

```diff
@@ -174,7 +230,26 @@ func (c *Client) Report(ctx context.Context, log *logrus.Entry, pj *v1.ProwJob)
 			log.WithError(err).Debug("Could not find PR commit, skipping retries")
 			err = nil
 		}
+		// Always return when there is any error reporting status context.
+		return []*v1.ProwJob{pj}, nil, err
+	}
+
+	// Check if this org or repo has opted out of failure report comments
+	refs := pj.Spec.Refs
+	fullRepo := fmt.Sprintf("%s/%s", refs.Org, refs.Repo)
+	for _, ident := range c.config().GitHubReporter.NoCommentRepos {
```

Comment:
Please add a comment that mentions this check has to be here and not in `ShouldReport` because we always need to report the status context, just potentially not create a comment